In [1]:
import pandas as pd
import datetime as dt

base = pd.read_csv("C:\\Users\jpvch\\Acidentes_DadosAbertos_20230912.csv", delimiter = ";")

C:\Users\jpvch\AppData\Local\Temp\ipykernel_12712\1142893163.py:4: DtypeWarning: Columns (15) have mixed types. Specify dtype option on import or set low_memory=False.
  base = pd.read_csv("C:\\Users\jpvch\\Acidentes_DadosAbertos_20230912.csv", delimiter = ";")


Importei as bibliotecas pandas e datetime, e as chamei de pd e dt, respectivamente. 
Comecei carregando os dados da base de dados.
Utilizei o método pd.read_csv() para ler o arquivo e armazená-lo em um DataFrame chamado base.

In [2]:
base.head()

num_acidente   chv_localidade data_acidente uf_acidente  ano_acidente  \
0       2448623  AC1200401201801    2018-01-24          AC          2018   
1       3477494  AC1200385201801    2018-01-06          AC          2018   
2       2939219  AC1200401201801    2018-01-04          AC          2018   
3         17595  AC1200203201801    2018-01-22          AC          2018   
4         58010  AC1200401201801    2018-01-21          AC          2018   

   mes_acidente  mes_ano_acidente  codigo_ibge     dia_semana fase_dia  ...  \
0             1             12018      1200401   QUARTA-FEIRA    MANHA  ...   
1             1             12018      1200385         SABADO    TARDE  ...   
2             1             12018      1200401   QUINTA-FEIRA    MANHA  ...   
3             1             12018      1200203  SEGUNDA-FEIRA    NOITE  ...   
4             1             12018      1200401        DOMINGO    MANHA  ...   

  lim_velocidade       tp_pista  ind_guardrail  ind_cantcentral  \
0  NAO INFORMADO  NAO INFORMADO  NAO INFORMADO    NAO INFORMADO   
1  NAO INFORMADO  NAO INFORMADO  NAO INFORMADO    NAO INFORMADO   
2  NAO INFORMADO  NAO INFORMADO  NAO INFORMADO    NAO INFORMADO   
3  NAO INFORMADO  NAO INFORMADO  NAO INFORMADO    NAO INFORMADO   
4  NAO INFORMADO  NAO INFORMADO  NAO INFORMADO    NAO INFORMADO   

   ind_acostamento qtde_acidente  qtde_acid_com_obitos  qtde_envolvidos  \
0    NAO INFORMADO             1                     0                3   
1    NAO INFORMADO             1                     0                1   
2    NAO INFORMADO             1                     0                2   
3    NAO INFORMADO             1                     0                2   
4    NAO INFORMADO             1                     0                2   

   qtde_feridosilesos  qtde_obitos  
0                   3            0  
1                   1            0  
2                   2            0  
3                   2            0  
4                   2            0  

[5 rows x 35 columns]

Depois, analisei quais eram os 5 primeiros registros da base com o método base.head()

In [3]:
base.tail()

num_acidente   chv_localidade data_acidente uf_acidente  \
4925598       4163385  TO1720853202305    2023-05-03          TO   
4925599       3233251  TO1713304202305    2023-05-02          TO   
4925600        922960  TO1715754202305    2023-05-08          TO   
4925601       3435150  TO1720804202305    2023-05-06          TO   
4925602       4441387  TO1709500202305    2023-05-16          TO   

         ano_acidente  mes_acidente  mes_ano_acidente  codigo_ibge  \
4925598          2023             5             52023      1720853   
4925599          2023             5             52023      1713304   
4925600          2023             5             52023      1715754   
4925601          2023             5             52023      1720804   
4925602          2023             5             52023      1709500   

            dia_semana fase_dia  ... lim_velocidade       tp_pista  \
4925598   QUARTA-FEIRA    TARDE  ...  NAO INFORMADO  NAO INFORMADO   
4925599    TERCA-FEIRA    TARDE  ...  NAO INFORMADO  NAO INFORMADO   
4925600  SEGUNDA-FEIRA    TARDE  ...  NAO INFORMADO  NAO INFORMADO   
4925601         SABADO    NOITE  ...  NAO INFORMADO  NAO INFORMADO   
4925602    TERCA-FEIRA    NOITE  ...  NAO INFORMADO  NAO INFORMADO   

         ind_guardrail  ind_cantcentral  ind_acostamento qtde_acidente  \
4925598  NAO INFORMADO    NAO INFORMADO    NAO INFORMADO             1   
4925599  NAO INFORMADO    NAO INFORMADO    NAO INFORMADO             1   
4925600  NAO INFORMADO    NAO INFORMADO    NAO INFORMADO             1   
4925601  NAO INFORMADO    NAO INFORMADO    NAO INFORMADO             1   
4925602  NAO INFORMADO    NAO INFORMADO    NAO INFORMADO             1   

         qtde_acid_com_obitos  qtde_envolvidos  qtde_feridosilesos  \
4925598                     0                2                   2   
4925599                     0                2                   2   
4925600                     0                2                   2   
4925601                     0                3                   3   
4925602                     0                2                   2   

         qtde_obitos  
4925598            0  
4925599            0  
4925600            0  
4925601            0  
4925602            0  

[5 rows x 35 columns]

Depois, analisei quais eram os 5 últimos registros da base com o método base.tail().

In [4]:
base.shape

(4925603, 35)

Usei o base.shape para analisar qual o tamanho de registros e o número de colunas.

In [5]:
inicio = pd.to_datetime(base['data_acidente']).dt.date.min()
print(inicio)

2018-01-01


Usei o to_datetime() para ler a coluna data_acidente e date.min() para verificar qual a data de início da base de dados.

In [6]:
fim = pd.to_datetime(base['data_acidente']).dt.date.max()
print(fim)

2023-05-31


E dessa vez, usei novamente o to_datetime e usei date.max() para verificar a última data registrada na base de dados.

In [7]:
base.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4925603 entries, 0 to 4925602
Data columns (total 35 columns):
 #   Column                Dtype  
---  ------                -----  
 0   num_acidente          int64  
 1   chv_localidade        object 
 2   data_acidente         object 
 3   uf_acidente           object 
 4   ano_acidente          int64  
 5   mes_acidente          int64  
 6   mes_ano_acidente      int64  
 7   codigo_ibge           int64  
 8   dia_semana            object 
 9   fase_dia              object 
 10  tp_acidente           object 
 11  cond_meteorologica    object 
 12  end_acidente          object 
 13  num_end_acidente      int64  
 14  cep_acidente          int64  
 15  bairro_acidente       object 
 16  km_via_acidente       int64  
 17  latitude_acidente     float64
 18  longitude_acidente    float64
 19  hora_acidente         int64  
 20  tp_rodovia            object 
 21  cond_pista            object 
 22  tp_cruzamento         object 
 23  tp_pavi

Usei o método base.info() para verificar as colunas, o tipo do dado e o tamanho da base.

In [8]:
base.isnull().sum()

num_acidente                  0
chv_localidade                0
data_acidente                 0
uf_acidente                   0
ano_acidente                  0
mes_acidente                  0
mes_ano_acidente              0
codigo_ibge                   0
dia_semana                    0
fase_dia                      0
tp_acidente                   0
cond_meteorologica            0
end_acidente             157888
num_end_acidente              0
cep_acidente                  0
bairro_acidente         1933910
km_via_acidente               0
latitude_acidente       3639454
longitude_acidente      3639589
hora_acidente                 0
tp_rodovia                    0
cond_pista                    0
tp_cruzamento                 0
tp_pavimento                  0
tp_curva                      0
lim_velocidade                0
tp_pista                      0
ind_guardrail                 0
ind_cantcentral               0
ind_acostamento               0
qtde_acidente                 0
qtde_aci

Usei o método base.isnull().sum() para analisar quantas colunas tinham valores nulos. Possibilitando-me ideias de quais colunas devo utilizar em minha ánalise final.

In [9]:
colunas_categoricas = ['ano_acidente', 'mes_acidente', 'hora_acidente', 'qtde_acidente', 'qtde_acid_com_obitos', 'qtde_envolvidos', 'qtde_feridosilesos', 'qtde_obitos']
estatisticas = base[colunas_categoricas].describe()
print(estatisticas)

       ano_acidente  mes_acidente  hora_acidente  qtde_acidente  \
count  4.925603e+06  4.925603e+06   4.925603e+06      4925603.0   
mean   2.020274e+03  6.489344e+00   1.479713e+05            1.0   
std    1.484895e+00  3.465866e+00   1.126696e+05            0.0   
min    2.018000e+03  1.000000e+00   0.000000e+00            1.0   
25%    2.019000e+03  3.000000e+00   1.000000e+05            1.0   
50%    2.020000e+03  6.000000e+00   1.430000e+05            1.0   
75%    2.022000e+03  1.000000e+01   1.820000e+05            1.0   
max    2.023000e+03  1.200000e+01   9.999990e+05            1.0   

       qtde_acid_com_obitos  qtde_envolvidos  qtde_feridosilesos   qtde_obitos  
count          4.925603e+06     4.925603e+06        4.925603e+06  4.925603e+06  
mean           2.257815e-02     1.589358e+00        1.564654e+00  2.470439e-02  
std            1.485543e-01     9.791111e-01        9.847560e-01  1.738893e-01  
min            0.000000e+00     0.000000e+00        0.000000e+00  0.0000

Filtrei em uma lista quais colunas queria utilizar em minha análise, depois utilizei o método base[lista].describe() junto com a base filtrada pela lista, para analisar qual a média, total de registros, valores minímos e máximos de colunas numéricas.

E o que cada linha dessa significa? 
count: O número total de observações (ou seja, acidentes) para cada coluna.
mean: A média (ou valor médio) das observações para cada coluna.
std: O desvio padrão das observações para cada coluna, isso mede a quantidade de variação ou dispersão dos valores.
min: O valor mínimo encontrado para cada coluna.
25%: O primeiro quartil para cada coluna, 25% das observações são menores que esse valor.
50%: A mediana para cada coluna, 50% das observações são menores que esse valor.
75%: O terceiro quartil para cada coluna, 75% das observações são menores que esse valor.
max: O valor máximo encontrado para cada coluna.

ano_acidente:
Contagem: 4,925,603
Média: 2020.27
Desvio padrão: 1.48
Mínimo: 2018
25% (primeiro quartil): 2019
50% (mediana): 2020
75% (terceiro quartil): 2022
Máximo: 2023

mes_acidente:
Contagem: 4,925,603
Média: 6.49
Desvio padrão: 3.47
Mínimo: 1
25% (primeiro quartil): 3
50% (mediana): 6
75% (terceiro quartil): 10
Máximo: 12

hora_acidente:
Contagem: 4,925,603
Média: 147,971.3
Desvio padrão: 112,669.6
Mínimo: 0
25% (primeiro quartil): 100,000
50% (mediana): 143,000
75% (terceiro quartil): 182,000
Máximo: 999,999

qtde_acidente:
Contagem: 4,925,603
Média: 1
Desvio padrão: 0
Mínimo: 1
25% (primeiro quartil): 1
50% (mediana): 1
75% (terceiro quartil): 1
Máximo: 1


qtde_acid_com_obitos:
Contagem: 4,925,603
Média: 0.0226
Desvio padrão: 0.1486
Mínimo: 0
25% (primeiro quartil): 0
50% (mediana): 0
75% (terceiro quartil): 0
Máximo: 1

qtde_envolvidos:
Contagem: 4,925,603
Média: 1.589
Desvio padrão: 0.9791
Mínimo: 0
25% (primeiro quartil): 1
50% (mediana): 2
75% (terceiro quartil): 2
Máximo: 67

qtde_feridosilesos:
Contagem: 4,925,603
Média: 1.565
Desvio padrão: 0.9848
Mínimo: 0
25% (primeiro quartil): 1
50% (mediana): 2
75% (terceiro quartil): 2
Máximo: 67

qtde_obitos:
Contagem: 4,925,603
Média: 0.0247
Desvio padrão: 0.1739
Mínimo: 0
25% (primeiro quartil): 0
50% (mediana): 0
75% (terceiro quartil): 0
Máximo: 42

Por fim, usei o método .value_counts(),que é um método que mostra os valores registrados das colunas. Dessa forma, posso ver quais são os valores registrados em cada coluna. Assim, foi possível eu decidir quais colunas usar, conforme meu ponto de vista.

In [13]:
base.num_acidente.value_counts()

num_acidente
2448623    1
1677022    1
935891     1
477055     1
4552496    1
          ..
3606690    1
2443535    1
1800510    1
4450370    1
4441387    1
Name: count, Length: 4925603, dtype: int64

In [14]:
base.chv_localidade.value_counts()

chv_localidade
MG3106200201911    7215
MG3106200201909    7192
MG3106200201912    7163
MG3106200201910    7142
MG3106200201905    7095
                   ... 
PE2609501202203       1
GO5213772202006       1
GO5210562202006       1
PE2612109202203       1
TO1720804202305       1
Name: count, Length: 208517, dtype: int64

In [15]:
base.data_acidente.value_counts()

data_acidente
2021-12-17    3898
2020-03-06    3857
2021-05-07    3835
2019-09-06    3807
2021-12-18    3793
              ... 
2023-04-21    1019
2023-04-23     996
2020-03-22     993
2023-04-07     953
2023-05-01     892
Name: count, Length: 1977, dtype: int64

In [16]:
base.uf_acidente.value_counts()

uf_acidente
MG    1369590
SP     754456
SC     632685
PR     485239
GO     467188
ES     200232
DF     176100
PA     144479
MS      99570
RJ      95805
CE      75143
BA      71324
PE      69654
RO      57796
PI      55981
RN      28059
MT      25395
TO      23193
AC      20812
MA      15251
RR      13254
PB      11323
AP       7437
RS       7310
AL       7208
AM       6763
SE       4356
Name: count, dtype: int64

In [17]:
base.ano_acidente.value_counts()

ano_acidente
2022    1023173
2021    1012768
2019    1007653
2020     912288
2018     721879
2023     247842
Name: count, dtype: int64

In [18]:
base.mes_acidente.value_counts()

mes_acidente
5     434032
3     427084
10    424952
2     424093
12    411459
11    411268
8     409887
1     406068
9     401959
4     396594
7     394947
6     383260
Name: count, dtype: int64

In [19]:
base.mes_ano_acidente.value_counts()

mes_ano_acidente
102022    93542
122021    93528
52022     91281
72021     91250
82022     90552
          ...  
22018     51923
42020     51237
12023     47118
52023     44645
42023     43097
Name: count, Length: 65, dtype: int64

In [20]:
base.codigo_ibge.value_counts()

codigo_ibge
3106200    380055
5300108    176100
3550308    173539
5208707    162099
4106902    138107
            ...  
4306379         1
4305835         1
4314175         1
4304853         1
1704105         1
Name: count, Length: 5469, dtype: int64

In [21]:
base.dia_semana.value_counts()

dia_semana
SEXTA-FEIRA      800737
SABADO           744681
SEGUNDA-FEIRA    712895
QUINTA-FEIRA     699731
QUARTA-FEIRA     690943
TERCA-FEIRA      678063
DOMINGO          598553
Name: count, dtype: int64

In [22]:
base.fase_dia.value_counts()

fase_dia
TARDE            1833981
MANHA            1371277
NOITE            1335740
MADRUGADA         330614
NAO INFORMADO      50657
DESCONHECIDO        3334
Name: count, dtype: int64

In [23]:
base.tp_acidente.value_counts()

tp_acidente
DESCONHECIDO                                 1183867
COLISAO                                       789694
COLISAO LATERAL                               484260
CHOQUE                                        449149
COLISAO TRASEIRA                              412645
OUTROS ACIDENTES DE TRANSITO                  398605
COLISAO FRONTAL                               303175
COLISAO TRANSVERSAL                           211218
QUEDA                                         176732
NAO INFORMADO                                 162567
ATROPELAMENTO COM PEDESTRE                    145423
CAPOTAMENTO                                    68490
TOMBAMENTO                                     63141
ATROPELAMENTO COM ANIMAIS                      41387
ENGAVETAMENTO                                  34652
ACIDENTE PESSOAL DE TRANSITO COM PEDESTRE        598
Name: count, dtype: int64

In [24]:
base.cond_meteorologica.value_counts()

cond_meteorologica
NAO INFORMADO                2562782
CLARO                         976279
DESCONHECIDAS                 841239
OUTRAS CONDICOES              432311
CHUVA                          66855
NUBLADO                        27167
GAROACHUVISCO                  15341
NEVOEIRO  NEVOA OU FUMACA       3507
VENTOS FORTES                    102
NEVE                              13
GRANIZO                            7
Name: count, dtype: int64

In [25]:
base.end_acidente.value_counts()

end_acidente
GERAL                                    56508
PRINCIPAL  RUA                           40692
ANEL RODOVIARIO                          19532
BR 381                                   19469
LOGRADOURO NAO INFORMADO                 19247
                                         ...  
RUA DELFINO BRASIL  CEP 74989-130            1
RUA MAJOR M ALVES                            1
RUA RI-02                                    1
AVENIDA ANHANGUERA ESQUINA COM RUA 14        1
AVENIDA GOIAS COM RUA D CENTRO               1
Name: count, Length: 702715, dtype: int64

In [26]:
base.num_end_acidente.value_counts()

num_end_acidente
0        3384981
1          78173
100        59183
10         37243
1000        7885
          ...   
54882          1
95645          1
52249          1
52175          1
96310          1
Name: count, Length: 59468, dtype: int64

In [27]:
base.cep_acidente.value_counts()

cep_acidente
0           4764717
61700000       1045
61760000        944
79900000        880
79750000        667
             ...   
37022595          1
35522170          1
35661010          1
35300296          1
95785000          1
Name: count, Length: 28849, dtype: int64

In [28]:
base.bairro_acidente.value_counts()

bairro_acidente
CENTRO                     362491
ZONA RURAL                  96359
RURAL                       41699
BAIRRO NAO IDENTIFICADO     34575
BRASILIA                    33663
                            ...  
ZONAS RURALS                    1
COOPACEM                        1
RESIDENCIAL VALENCIA 1          1
COPA LIZ                        1
SETOR SUSSUAPARA                1
Name: count, Length: 37288, dtype: int64

In [30]:
base.km_via_acidente.value_counts()

km_via_acidente
0       4646068
1         10295
2          4959
10         4841
4          4534
         ...   
1883          1
3327          1
2443          1
3413          1
4607          1
Name: count, Length: 3915, dtype: int64

In [32]:
base.latitude_acidente.value_counts()

latitude_acidente
-0.000000     356963
-20.355800      4235
-20.127100      4085
-20.299500      4000
-20.848200      2747
               ...  
-19.899774         1
-17.854498         1
-20.491483         1
-19.936583         1
-11.735969         1
Name: count, Length: 515152, dtype: int64

In [33]:
base.longitude_acidente.value_counts()

longitude_acidente
-0.000000     356962
-40.302400      4235
-40.305400      4085
-40.313200      4000
-41.130800      2841
               ...  
-44.257754         1
-42.577309         1
-43.864273         1
-44.341833         1
-49.076847         1
Name: count, Length: 516084, dtype: int64

In [34]:
base.hora_acidente.value_counts()

hora_acidente
0         70930
170000    67363
180000    66877
160000    65122
999999    64661
          ...  
11457         1
41458         1
35412         1
22430         1
51709         1
Name: count, Length: 83923, dtype: int64

In [35]:
base.tp_rodovia.value_counts()

tp_rodovia
NAO INFORMADO    3436167
MUNICIPAL        1102292
ESTADUAL          168143
DESCONHECIDO      145732
FEDERAL            73269
Name: count, dtype: int64

In [15]:
base.cond_pista.value_counts()

cond_pista
NAO INFORMADO             3478376
SECA                       979776
DESCONHECIDO               344363
MOLHADA                    115158
ESCORREGADIA                 3402
OBSTRUIDA                    1718
COM BURACO                   1520
COM LAMA                      870
COM MATERIAL GRANULADO        420
Name: count, dtype: int64

In [16]:
base.tp_pavimento.value_counts()

tp_pavimento
NAO INFORMADO     3278653
ASFALTO           1309041
DESCONHECIDO       291176
TERRA               18961
PARALELEPIPEDO      14669
CONCRETO             7339
CASCALHO             5764
Name: count, dtype: int64

In [36]:
base.tp_curva.value_counts()

tp_curva
NAO INFORMADO      4006528
DESCONHECIDO        821420
NAO HAVIA CURVA      81886
CURVA ABERTA         11653
CURVA ACENTUADA       4116
Name: count, dtype: int64

In [37]:
base.lim_velocidade.value_counts()

lim_velocidade
NAO INFORMADO       4818779
40 KMH                40419
60 KMH                34118
80 KMH                16265
30 KMH                11585
110 KMH                4435
ACIMA DE 110 KMH          2
Name: count, dtype: int64

In [38]:
base.tp_pista.value_counts()

tp_pista
NAO INFORMADO             4542285
SIMPLES  COM CICLOVIA      136355
SIMPLES  SEM CICLOVIA      113929
DUPLA  SEM CICLOVIA         97129
DESCONHECIDO                34047
MULTIPLA  SEM CICLOVIA       1539
MULTIPLA  COM CICLOVIA        172
DUPLA  COM CICLOVIA           147
Name: count, dtype: int64

In [39]:
base.ind_guardrail.value_counts()

ind_guardrail
NAO INFORMADO    4441154
DESCONHECIDO      456403
NAO                27876
SIM                  170
Name: count, dtype: int64

In [40]:
base.ind_cantcentral.value_counts()

ind_cantcentral
NAO INFORMADO    4168066
DESCONHECIDO      455925
SIM               152696
NAO               148916
Name: count, dtype: int64

In [41]:
base.ind_acostamento.value_counts()

ind_acostamento
NAO INFORMADO    3572365
DESCONHECIDO      881930
SIM               294466
NAO               176842
Name: count, dtype: int64

In [42]:
base.qtde_acidente.value_counts()

qtde_acidente
1    4925603
Name: count, dtype: int64

In [43]:
base.qtde_acid_com_obitos.value_counts()

qtde_acid_com_obitos
0    4814392
1     111211
Name: count, dtype: int64

In [44]:
base.qtde_envolvidos.value_counts()

qtde_envolvidos
1     1977629
2     1948397
0      430261
3      417877
4      101282
5       30664
6       10806
7        4249
8        1882
9         919
10        500
11        234
12        200
13        124
14        110
15         59
16         57
17         44
19         37
20         32
18         31
21         20
22         18
25         15
23         14
27         12
24         11
31         10
26          8
42          8
37          8
30          7
28          7
34          7
29          6
35          6
40          5
38          5
32          5
41          4
46          4
33          3
39          3
44          3
56          2
36          2
45          2
61          2
50          1
48          1
58          1
54          1
43          1
52          1
67          1
55          1
51          1
66          1
53          1
49          1
Name: count, dtype: int64

In [45]:
base.qtde_feridosilesos.value_counts()

qtde_feridosilesos
1     1932846
2     1931551
0      502384
3      412609
4       98880
5       29396
6       10024
7        3909
8        1706
9         824
10        428
11        204
12        173
13        119
14         99
15         61
16         57
17         37
19         37
18         34
20         28
21         21
23         17
22         14
25         14
37         12
27         10
24          9
26          9
28          9
31          8
35          7
40          6
42          5
32          5
29          5
34          5
33          4
46          4
38          4
44          3
41          3
49          3
30          3
54          2
39          2
36          2
51          2
58          1
55          1
45          1
53          1
48          1
67          1
62          1
60          1
61          1
Name: count, dtype: int64

In [46]:
base.qtde_obitos.value_counts()

qtde_obitos
0     4814392
1      103074
2        6674
3         966
4         308
5         118
6          37
7          18
8           6
10          3
12          2
15          2
11          1
9           1
42          1
Name: count, dtype: int64